client/src/main/java/io/split/telemetry/storage/InMemoryTelemetryStorage.java

```diff
@@ -209,6 +227,14 @@ public long getSessionLength() {
         return _sdkRecords.get(SdkRecordsEnum.SESSION).get();
     }
 
+    @Override
+    public UpdatesFromSSE popUpdatesFromSSE() {
+        UpdatesFromSSE updatesFromSSE = new UpdatesFromSSE();
+        updatesFromSSE.setSplits(_updatesFromSSERecords.get(UpdatesFromSSEEnum.SPLITS).get());
+        _updatesFromSSERecords.replace(UpdatesFromSSEEnum.SPLITS, new AtomicLong());
```

Comment:
This way the AtomicLong value can be retrieved and updated atomically
```suggestion
        updatesFromSSE.setSplits(_updatesFromSSERecords.get(UpdatesFromSSEEnum.SPLITS).getAndSet(0L));
```

Reply:
Updated.